src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java

```diff
@@ -95,7 +95,7 @@
 
         this.samlRolesKey = settings.get("roles_key");
         this.samlSubjectKey = settings.get("subject_key");
-        this.samlRolesSeparator = settings.get("roles_seperator");
+        this.samlRolesSeparator = settings.get("roles_separator");
```

Comment:
It should support both to support migration use cases as this is index

Reply:
By both, do you mean with the typo and without the typo?

Reply:
yes

Reply:
This is settings and the old setting needs to be deprecated.

Reply:
As far as I can see this setting was never documented with a typo. As @orsifacundo correctly pointed out in #969 [this doc](https://opendistro.github.io/for-elasticsearch-docs/docs/security/configuration/proxy/) has it spelled correctly. Do you still think we need to introduce backwards compatibility with the typoed version @cliu123 and @vrozov?

Reply:
I agree with @vrozov that the typoed one should be deprecated. If we decide to go with the changes we currently have in this PR, we can backport the changes to old version and let users decide if they want to migrate. If users decide to migarte, the typoed setting should be fixed before migration.

Reply:
I've created a new PR https://github.com/opensearch-project/security/pull/1618 that supports backwards compatibility.